editoast/src/views/timetable.rs

```diff
@@ -557,20 +560,77 @@ async fn conflicts(
     let conflicts = conflict_detection_response.conflicts;
     let conflicts_response: Result<Vec<Conflict>> = conflicts
         .into_iter()
-        .map(|conflict| Conflict::into_conflict_response(conflict, map_string_to_id.clone()))
+        .map(Conflict::from_core_response)
         .collect();
     Ok(Json(conflicts_response?))
 }
 
+async fn retrieve_trains_and_paced_trains(
+    conn: &mut DbConnection,
+    timetable_id: i64,
+) -> Result<(Vec<TrainSchedule>, Vec<models::PacedTrain>)> {
+    let timetable_trains = TimetableWithTrains::retrieve_or_fail(conn, timetable_id, || {
+        TimetableError::NotFound { timetable_id }
+    })
+    .await?;
+
+    let (trains, _): (Vec<_>, _) =
+        TrainSchedule::retrieve_batch(conn, timetable_trains.train_ids).await?;
+
+    let (paced_trains, _): (Vec<_>, _) =
+        models::PacedTrain::retrieve_batch(conn, timetable_trains.paced_train_ids).await?;
+
+    Ok((trains, paced_trains))
+}
+
+async fn retrieve_simulations(
+    db_pool: Arc<DbConnectionPoolV2>,
+    valkey_client: Arc<ValkeyClient>,
+    core_client: Arc<CoreClient>,
+    trains: &[TrainSchedule],
+    paced_trains: Vec<models::PacedTrain>,
+    infra: &Infra,
+    electrical_profile_set_id: Option<i64>,
+) -> Result<(
+    Vec<(SimulationResponse, PathfindingResult)>,
+    Vec<(SimulationResponse, PathfindingResult)>,
+)> {
+    let train_simulations = train_simulation_batch(
+        &mut db_pool.get().await?,
+        valkey_client.clone(),
+        core_client.clone(),
+        trains,
+        infra,
+        electrical_profile_set_id,
+    )
+    .await?;
+
+    let paced_train_to_ts = paced_trains
+        .into_iter()
+        .flat_map(|pt| pt.iter_occurrences())
+        .collect::<Vec<_>>();
+
+    let paced_train_simulations = train_simulation_batch(
+        &mut db_pool.get().await?,
+        valkey_client.clone(),
+        core_client.clone(),
+        &paced_train_to_ts,
+        infra,
+        electrical_profile_set_id,
+    )
+    .await?;
+
+    Ok((train_simulations, paced_train_simulations))
+}
+
 fn build_conflict_core_request(
     infra: Infra,
     trains: Vec<TrainSchedule>,
     train_simulations: Vec<(SimulationResponse, PathfindingResult)>,
-    paced_trains_ts: Vec<TrainSchedule>,
+    paced_trains: Vec<models::PacedTrain>,
```

Comment:
```suggestion
    paced_trains: &[models::PacedTrain],
```

we can avoid taking ownership in functions where this is not necessary